### ISSUE TITLE
[Bug] Modification of a MM2 Resource doesn't update ReplicaSet

### ISSUE THREAD

**Issue by AljoschaP** [strimzi/strimzi-kafka-operator#3303]
**Describe the bug**
A modification of a Mirrormaker2 resource doesn't force a change in the related replicaset which in turn doesn't update and rollout the related pods.
**To Reproduce**
1) Deploy MM2
2) Change a field of the resource, e.g. 
```
template:
    pod:
      imagePullSecrets:
        - name: gitlab-pull-secrets
```
3) You can see the difference between the definition of the MM2 resource and the related replicaset but no reconciliation takes place

**Expected behavior**
A reconciliation takes place, the replicaset is updated and a rollout happens which restart and update the pods.

**Environment (please complete the following information):**
 - Strimzi version: 0.18.0
 - Installation method: YAML
 - Kubernetes cluster: Kubernetes 1.16.9
 - Infrastructure: Azure AKS

**YAML files and logs**

-

**Additional context**
-

**Comment by scholzj** [comment]
I tried this on my cluster (Kube 1.18.5) and it seems to work fine.
* I deployed the MM2 from examples
* After it was ready, I edited it to add the pull secret
* The Cluster Operator updated the deployment which triggered new replica set and new pod and scaled down the old one after it was ready.

So it seems to work fine for me. 

TBH, Strimzi is using regular deployment for Mirror Maker 2. So we just update the deployment and the rest is up to the Kubernetes Deployment Controller. So if the deployment has the new secret but the replica set is not created it looks more like some Kubrnetes issue.

**Comment by kramik1** [comment]
I know this is closed but I didn't see a topic like this in discussion. I updated my source IP in the type mirrormaker2 strimzi file and pushed the changes. It didn't do anything. And then I noticed there was a deployment for mirrormaker2 like you mentioned here. So I cycled that as well. The mirrormaker2 instances do not pick up the new source IP. I also tried killing the pods as a last ditch effort before deleting the mirrormaker2 instance altogether and redeploying. I am using 0.21.1. It is still picking up the old IPs.

**Comment by scholzj** [comment]
You will need to share some logs and CRs I guess. Might be also worth trying if it does the same with 0.22. But I'm not sure the source IP address is part of the Pod declaration it self - my guess is that it is configured dynamically over the Connect REST API.. So I'm not sure you should expect the operator to roll the pod - so this might be different from this issue.

**Comment by kramik1** [comment]
Thanks for the response. I had to rename the yaml config instance to get it to deploy with the new IP. At first I had a stray character in the yaml file (operator exceptions). But when I fixed it, it didn't make a difference. I think you are right about the IP being related to the connect API. I misspoke before, I upgrade to 0.22.1 already with a clean install. I am still developing the solution, we are not in production yet. If I run into this again I will open a bug report and try and get the logs etc.
